context/src/main/java/io/opentelemetry/context/ContextStorageWrappers.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Holder of functions to wrap the used {@link ContextStorage}. Separate class from {@link
+ * LazyStorage} to allow registering wrappers before initializing storage.
+ */
+final class ContextStorageWrappers {
+
+  private static final Logger log = Logger.getLogger(ContextStorageWrappers.class.getName());
+
+  private static boolean storageInitialized;
+
+  private static final List<Function<? super ContextStorage, ? extends ContextStorage>> wrappers =
+      new ArrayList<>();
+
+  static synchronized void addWrapper(
```

Comment:
It probably doesn't matter too much, since this class isn't public, but exposing synchronized methods is generally a bad practice, as it could allow accidental deadlocking if someone synchronizes on this class.

Reply:
Yeah figured it's ok due to the privateness but went with consistency.